logstash-core/lib/logstash/instrument/collector.rb

```diff
@@ -1,79 +1,109 @@
 # encoding: utf-8
 require "logstash/instrument/snapshot"
+require "logstash/instrument/metric_store"
 require "logstash/util/loggable"
-require "concurrent/map"
+require "concurrent/timer_task"
 require "observer"
 require "singleton"
 require "thread"
 
 module LogStash module Instrument
+  # The Collector singleton is the single point of reference for all
+  # the metrics collection inside logstash, the metrics library will make
+  # direct calls to this class.
+  #
+  # This class is an observable responsable of periodically emitting view of the system
+  # to other components like the internal metrics pipelines.
   class Collector
-    # TODO: move to a timer based  API with concurrent ruby timer class
-    # Allow to accept an external flush
-    # When the flush is done we should not record any new metric
     include LogStash::Util::Loggable
     include Observable
     include Singleton
 
-    SNAPSHOT_ROTATION_TIME = 1 #seconds
+    SNAPSHOT_ROTATION_TIME = 1 # seconds
```

Comment:
I always have troubles with the units and vars/constants? any idea on how to make this less error prone?